### DISCUSSION TITLE
What will we do differently in Lagoon 3

### DISCUSSION THREAD

**Discussion by tobybellwood** [amazeeio/lagoon#2672] [Roadmap]
With Lagoon 2 _almost_ out the door, lets start turning our eyes towards the next major release.

What should be in scope - what do we need to change, what should we change, what can we look at next?

To start the ball rolling, we've covered these in the past, maybe now is the time?

- API rewrite
- Notifications system
- Build process
- Where should Auth live?

**Comment by smlx** [comment]
> Where should Auth live?

We might need to be a bit more specific since these might not be in the same place:
* Where should Authentication live?
* Where should Authorization live?

Some more ideas:
* DB schema migrations and versioning for Lagoon Core
* Support DBaaS for Lagoon Core
* decentralised SSH

**Comment by tobybellwood** [comment]
* api-db migrate to kubernetes custom-resources?
* ability to have different environments hosted in different kubernetes clusters?

**Reply by tobybellwood** [reply]
both of these have been implemented now :partying_face:
